backends/qualcomm/runtime/QnnManager.cpp

```diff
@@ -680,7 +680,30 @@ Error QnnManager::Compile(
         QNN_GET_ERROR_CODE(error));
     return Error::Internal;
   }
+  
+  auto qnn_profile_level =
+      backend_params_ptr_->qnn_graph_ptr_->GetProfile()->GetQnnProfileLevel();
+  // Write context binary to file for debugging purpose
+  if (qnn_profile_level == QNN_PROFILE_LEVEL_DETAILED) {
+    auto ret = backend_params_ptr_->qnn_context_ptr_->GetContextBinary(
+            qnn_executorch_context_binary);
+    if (ret != Error::Ok) {
+        QNN_EXECUTORCH_LOG_WARN("Fail to get context binary.");
+        return Error::Ok;
+    }
+    static const char* binary_file = "/tmp/qnn_context_binary.bin";
```

Comment:
How about we specify this with an environment variable?
It that environment variable is not empty, we dump context-bianries to that folder...etc.
Then we don't need to couple `qnn_profile_level` with this functionality.

Reply:
Good idea. Let me try this out.

Reply:
@chiwwang Updated to use env variable.